internal/api/handlers.go

```diff
@@ -0,0 +1,14 @@
+package api
+
+import (
+	"fmt"
+	"net/http"
+)
+
+func HealthCheckHandler(w http.ResponseWriter, r *http.Request) {
```

Comment:
I think this function doesn't have to be exported as it lives and is used in the same package.
```suggestion
func healthCheckHandler(w http.ResponseWriter, r *http.Request) {
```